setup.py

```diff
@@ -52,26 +54,48 @@
 )
 
 
-py_version = sys.version[:3]
+py_version = sys.version_info
+is_pypy = "PyPy" in sys.version
 
 SCRIPT1 = 'nose2'
-SCRIPT2 = 'nose2-%s' % (py_version,)
+SCRIPT2 = 'nose2-%s.%s' % (py_version.major, py_version.minor)
+
 
+def _get_specific_requirements():
+    directory = dirname(__file__)
+    version_specific_requirements_path = 'requirements-{}.{}.txt'.format(py_version.major, py_version.minor)
+    if is_pypy:
+        pypy_name = "pypy"
+        if py_version.major == 3:
+            pypy_name = 'pypy3'
+        version_specific_requirements_path = 'requirements-{}.txt'.format(pypy_name)
+    if not os.path.exists(os.path.join(directory, version_specific_requirements_path)):
+
+        if py_version.major == 3:
+            version_specific_requirements_path = 'requirements-3.5.txt'
+        else:
+            version_specific_requirements_path = 'requirements-2.7.txt'
+    with open(os.path.join(directory, version_specific_requirements_path)) as requirements_file:
+        REQS.update(list(filter(lambda s: s and not s.startswith('#'), requirements_file.readlines())))
+    with open(os.path.join(directory, 'requirements.txt')) as requirements_file:
+        REQS.update(list(filter(lambda s: s and not s.startswith('#'), requirements_file.readlines())))
+    print(REQS)
 try:
     from setuptools import setup
 except ImportError:
     from distutils.core import setup
 else:
 
-    REQS = ['six>=1.1']
-
+    REQS = set()
+    _get_specific_requirements()
     params['entry_points'] = {
         'console_scripts': [
             '%s = nose2:discover' % SCRIPT1,
             '%s = nose2:discover' % SCRIPT2,
         ],
     }
-    params['install_requires'] = REQS
+    params['install_requires'] = list(REQS)
     params['test_suite'] = 'unittest.collector'
+    params['extras_require']['doc'] = ['Sphinx>=1.0.5']
```

Comment:
so, [this answer](https://stackoverflow.com/questions/19559247/requirements-txt-depending-on-python-version) suggests using this value to solve this whole problem. Not sure if it will work with `pypy` but maybe worth a shot?

Reply:
I did not knew it. I think I will make a full rework.